pgsql-flexible-server-db.tf

```diff
@@ -1,8 +1,11 @@
 resource "azurerm_postgresql_flexible_server_database" "pg_databases" {
-  for_each = var.pgsql_databases
+  for_each = {
+    for index, db in var.pgsql_databases :
+    db.name => db
+  }
 
   name      = each.value.name
   server_id = azurerm_postgresql_flexible_server.pgsql_server.id
-  collation = each.value.collation
-  charset   = each.value.charset
+  collation = try(each.value.collation, null)
+  charset   = try(each.value.charset, null)
```

Comment:
don't force the user to set the values, we just fallback to the sensible defaults